structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/rest/filter/CDPStructuredEventFilter.java

```diff
@@ -0,0 +1,383 @@
+            requestContext.setEntityStream(logInboundEntity(body, requestContext.getEntityStream(), MessageUtils.getCharset(requestContext.getMediaType())));
+            requestContext.setProperty(REST_PARAMS, getRequestUrlParameters(requestContext));
+            requestContext.setProperty(REQUEST_DETAILS, createRequestDetails(requestContext, body.toString()));
+        }
+    }
+
+    @Override
+    public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext) {
+        if (BooleanUtils.isTrue((Boolean) requestContext.getProperty(LOGGING_ENABLED_PROPERTY))) {
+            RestResponseDetails restResponse = createResponseDetails(responseContext);
+            if (responseContext.hasEntity()) {
+                OutputStream stream = new LoggingStream(responseContext.getEntityStream());
+                responseContext.setEntityStream(stream);
+                requestContext.setProperty(LOGGINGSTREAM_PROPERTY, stream);
+                requestContext.setProperty(RESPONSE_DETAILS, restResponse);
+            } else {
+                Long requestTime = (Long) requestContext.getProperty(REQUEST_TIME);
+                RestRequestDetails restRequest = (RestRequestDetails) requestContext.getProperty(REQUEST_DETAILS);
+                Map<String, String> restParams = (Map<String, String>) requestContext.getProperty(REST_PARAMS);
+                sendStructuredEvent(restRequest, restResponse, restParams, requestTime, "");
+            }
+        }
+    }
+
+    @Override
+    public void aroundWriteTo(WriterInterceptorContext context) throws IOException, WebApplicationException {
+        context.proceed();
+        if (BooleanUtils.isTrue((Boolean) context.getProperty(LOGGING_ENABLED_PROPERTY))) {
+            Long requestTime = (Long) context.getProperty(REQUEST_TIME);
+            RestRequestDetails restRequest = (RestRequestDetails) context.getProperty(REQUEST_DETAILS);
+            RestResponseDetails restResponse = (RestResponseDetails) context.getProperty(RESPONSE_DETAILS);
+            String responseBody = ((LoggingStream) context.getProperty(LOGGINGSTREAM_PROPERTY)).getStringBuilder(
+                    MessageUtils.getCharset(context.getMediaType())).toString();
+            Map<String, String> restParams = (Map<String, String>) context.getProperty(REST_PARAMS);
+            if (restParams == null) {
+                restParams = new HashMap<>();
+            }
+            sendStructuredEvent(restRequest, restResponse, restParams, requestTime, responseBody);
+        }
+    }
+
+    private void sendStructuredEvent(RestRequestDetails restRequest, RestResponseDetails restResponse, Map<String, String> restParams, Long requestTime,
+            String responseBody) {
+        boolean valid = checkRestParams(restParams);
+        try {
+            if (!valid) {
+                LOGGER.debug("Cannot create structured event, because rest params are invalid.");
+                return;
+            }
```

Comment:
please don't return in the middle of the method, at least add an `else` branch

Reply:
I will refactor this `return;` in the follow-up jira.